convert2rhel_insights_tasks/main.py

```diff
@@ -877,6 +871,41 @@ def update_insights_inventory():
     logger.info("System registered with insights-client successfully.")
 
 
+def check_repos_are_valid():
+    """Check if the repositories under /etc/yum.repos.d are available.
+
+    :raises ProcessExit: In case any of the repositories defined in that folder
+        is not available.
+    """
+    logger.info("Checking for system repositories accessbility")
+    output, return_code = run_subprocess(
+        cmd=["/usr/bin/yum", "makecache", "--setopt=*.skip_if_unavailable=False"]
```

Comment:
I tried this command and have some interesting obresvation.

When this command is run with correctly set repos, then the repos are set to inaccessible values (e.g. changing `vault` to `mirror`), the return value is 0 even with inaccesible repos:
```
# yum makecache --setopt=*.skip_if_unavailable=False
Loaded plugins: fastestmirror, product-id, search-disabled-repos, subscription-manager

This system is registered with an entitlement server, but is not receiving updates. You can use subscription-manager to assign subscriptions.

Loading mirror speeds from cached hostfile
http://mirror.centos.org/7.9.2009/os/x86_64/repodata/repomd.xml: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below wiki article 

https://wiki.centos.org/yum-errors

If above article doesn't help to resolve this issue please use https://bugs.centos.org/.

client_tools_rhel7                                                                                                                                         | 3.8 kB  00:00:00     
copr:copr.fedorainfracloud.org:group_oamg:rhc-worker-script                                                                                                | 1.5 kB  00:00:00     
extras                                                                                                                                                     | 3.6 kB  00:00:00     
ubi_rhel7                                                                                                                                                  | 3.8 kB  00:00:00     
updates                                                                                                                                                    | 3.6 kB  00:00:00     
Metadata Cache Created
# echo $?
0
```

When I run `yum clean metadata` (haven't tested all the options, but metadata looked as least invazive one) before it, this command `# yum makecache --setopt=*.skip_if_unavailable=False` returns 1 if there is any unavailable repo. So there are some leftovers in the yum cache.

Reply:
@hosekadam, good point! When introducing the `yum makecache` command in convert2rhel to reveal inaccessible repos we knew that the cache needs to be cleared before that. That's why convert2rhel does clear yum cache early https://github.com/oamg/convert2rhel/blob/main/convert2rhel/pkgmanager/__init__.py#L87. We need to do the same here.

Reply:
Good catch, folks! Thanks for that. I will update the code